elastic_transport/client_utils.py

```diff
@@ -149,22 +149,15 @@ def to_bytes(
     return value
 
 
-# Python 3.7 added '~' to the safe list for urllib.parse.quote()
-_QUOTE_ALWAYS_SAFE = frozenset(
-    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_.-~"
-)
-
-
 def percent_encode(
     string: Union[bytes, str],
     safe: str = "/",
     encoding: Optional[str] = None,
     errors: Optional[str] = None,
 ) -> str:
     """Percent-encodes a string so it can be used in an HTTP request target"""
-    # Redefines 'urllib.parse.quote()' to always have the '~' character
-    # within the 'ALWAYS_SAFE' list. The character was added in Python 3.7
-    safe = "".join(_QUOTE_ALWAYS_SAFE.union(set(safe)))
+    # This function used to add `~` to unreserverd characters, but this was fixed in Python 3.7.
+    # Keeping the function here as it is part of the public API.
```

Comment:
Can the function be deprecated, though? My guess is that we'd want to steer people away from using a custom function now that it is simply a wrapper.

Reply:
It can be deprecated, but is currently used in elasticsearch-py. I'll look into doing it, thanks.

Reply:
Opened https://github.com/elastic/elasticsearch-py/pull/2415